crates/primitives/src/transaction/mod.rs

```diff
@@ -612,6 +614,93 @@ impl From<TxEip4844> for Transaction {
     }
 }
 
+impl TryFrom<reth_rpc_types::Transaction> for Transaction {
+    type Error = ConversionError;
+
+    fn try_from(tx: reth_rpc_types::Transaction) -> Result<Self, Self::Error> {
+        match tx.transaction_type {
+            None => {
+                // legacy
+                Ok(Transaction::Legacy(TxLegacy {
+                    chain_id: tx.chain_id,
+                    nonce: tx.nonce,
+                    gas_price: tx.gas_price.ok_or(ConversionError::MissingGasPrice)?,
```

Comment:
this is a bit dangerous, because gasprice is commonly set for all tx types, so this match arm should check if there are any 1559 fields present, if so return an error